custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -16,8 +16,14 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
             }
         };
         vm.selectedLocationsCount = function() {
-              return (vm.selectedLocationLevel===4) ? (vm.selectedLocations.filter(Boolean).length) :
-                  (vm.selectedLocations.filter(Boolean).length-1);
+            // this method returns selectedLocationLevel
+            // TODO: Need to check why selectedLocationLevel is undefined for all location levels except when awc is selected
+            // when awc is selected return selected location level as is or count non null elements in selected Locations array
+            // selectedLocations array is fixed array of size 5. default value: [national, null,null,null,null]
+            // vm.selectedLocations.filter(Boolean) returns array after removing all null entries
+            // eg: if block is selected selectedLocations is (state, district, block, national, null)- only 3 levels selected.
```

Comment:
should this be `(state, district, block, null, null)`? I'm confused by the "national" in the fourth position.

Reply:
or maybe `(national, state, district, block, null)`?

Reply:
Sorry, I got confused(its "all"  not "national"). There is nothing wrong with selectedLocations.  Array status in different scenarios:
no location selected: [all, null,null,null,null]
state selected: [state,all,null,null,null]
district selected: [state, district, all, null,null]...